images/cilium/Dockerfile

```diff
@@ -59,17 +59,20 @@ RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium \
 # Extract debug symbols to /tmp/debug and strip the binaries if NOSTRIP is not set.
 RUN set -xe && \
     export D=/tmp/debug/${TARGETOS}/${TARGETARCH} && \
-    mkdir -p $D && \
     cd /tmp/install/${TARGETOS}/${TARGETARCH} && \
     find . -type f \
       -executable \
       -exec sh -c \
         'filename=$(basename ${0}) && \
          objcopy --only-keep-debug ${0} ${0}.debug && \
          if ! echo "$MODIFIERS" | grep "NOSTRIP=1" ; then objcopy --strip-all ${0} && (cd $(dirname ${0}) && objcopy --add-gnu-debuglink=${filename}.debug ${filename}) ; fi && \
-         mv -v ${0}.debug ${D}/${filename}.debug' \
+         mkdir -p $(dirname ${D}/${0}) && \
+         mv -v ${0}.debug ${D}/${0}.debug' \
       {} \;
 
+# Check debug symbols are present
+RUN for f in $(find /tmp/debug -type f -name '*.debug') ; do readelf -S ${f} | grep -q \\.symtab ; done
```

Comment:
What happens if this check fails? Does it just silently break the build? It would be nice to make sure that an error in finding the debug symbols generates a clear message for any future developer who may stray to break this.

Reply:
Indeed, it’s a little cryptic, so I’ve made it fail more explicitly:

```
 $  docker build . -f images/cilium/Dockerfile --target builder -t builder:1
…
 => ERROR [builder 6/7] RUN for f in $(find /tmp/debug -type f -name '*.debug') ; do readelf -S ${f} | grep -q \.symtab ||     (echo Debug symbols are missing in ${f} - possibly due to incorrect build parameters && false); done                                        0.1s
------
 > [builder 6/7] RUN for f in $(find /tmp/debug -type f -name '*.debug') ; do readelf -S ${f} | grep -q \.symtab ||     (echo Debug symbols are missing in ${f} - possibly due to incorrect build parameters && false); done:
0.109 Debug symbols are missing in /tmp/debug/linux/arm64/opt/cni/bin/cilium-cni.debug - possibly due to incorrect build parameters
0.109 Debug symbols are missing in /tmp/debug/linux/arm64/usr/bin/cilium-agent.debug - possibly due to incorrect build parameters
0.110 Debug symbols are missing in /tmp/debug/linux/arm64/usr/bin/cilium-bugtool.debug - possibly due to incorrect build parameters
0.110 Debug symbols are missing in /tmp/debug/linux/arm64/usr/bin/cilium-health-responder.debug - possibly due to incorrect build parameters
0.111 Debug symbols are missing in /tmp/debug/linux/arm64/usr/bin/cilium-health.debug - possibly due to incorrect build parameters
0.112 Debug symbols are missing in /tmp/debug/linux/arm64/usr/bin/cilium-mount.debug - possibly due to incorrect build parameters
0.112 Debug symbols are missing in /tmp/debug/linux/arm64/usr/bin/cilium-sysctlfix.debug - possibly due to incorrect build parameters
0.113 Debug symbols are missing in /tmp/debug/linux/arm64/usr/bin/cilium-dbg.debug - possibly due to incorrect build parameters
------
Dockerfile:74
--------------------
  73 |     # Check debug symbols are present
  74 | >>> RUN for f in $(find /tmp/debug -type f -name '*.debug') ; do readelf -S ${f} | grep -q \\.symtab || \
  75 | >>>     (echo Debug symbols are missing in ${f} - possibly due to incorrect build parameters && false); done
  76 |
--------------------
ERROR: failed to solve: process "/bin/sh -c for f in $(find /tmp/debug -type f -name '*.debug') ; do readelf -S ${f} | grep -q \\\\.symtab ||     (echo Debug symbols are missing in ${f} - possibly due to incorrect build parameters && false); done" did not complete successfully: exit code: 1
```